staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -433,19 +454,25 @@ func (w ConditionalWait) checkCondition(obj *unstructured.Unstructured) (bool, e
 		return false, nil
 	}
 	for _, conditionUncast := range conditions {
-		condition := conditionUncast.(map[string]interface{})
-		name, found, err := unstructured.NestedString(condition, "type")
-		if !found || err != nil || !strings.EqualFold(name, w.conditionName) {
-			continue
-		}
-		status, found, err := unstructured.NestedString(condition, "status")
-		if !found || err != nil {
-			continue
+		for nameFromWait, statusFromWait := range w.conditions {
+			condition := conditionUncast.(map[string]interface{})
+			name, found, err := unstructured.NestedString(condition, "type")
+			if !found || err != nil || !strings.EqualFold(name, nameFromWait) {
+				continue
+			}
+			status, found, err := unstructured.NestedString(condition, "status")
+			if !found || err != nil {
+				continue
+			}
+
+			results = append(results, strings.EqualFold(status, statusFromWait))
 		}
-		return strings.EqualFold(status, w.conditionStatus), nil
 	}
-
-	return false, nil
+	final := false
+	for _, v := range results {
+		final = final || v
```

Comment:
Since this is an `OR`, shouldn't we be able to return from the loop as soon as we have a `true`?

Reply:
Yes but if there's only one value, the complexity remains O(1) and the `OR` operation isn't so compute expensive.

Reply:
Yes, my point isn't so much about performance but about readability. The fact that you save all the results in a list seem to imply that you have a greater goal for them :-)